chess/frontend/gui/src/modules/Game/Game.jsx

```diff
@@ -163,55 +176,88 @@ export default class Game extends React.Component {
   // to handle moving the pieces
   handleClick(index) {
-    this.checkPossibleMoves(index);
-    // If the user selects the oponent's peice first
-    if (
-      this.state.selected === NOT_SELECTED && this.state.turn !== this.state.squares[index].player) {
-      this.setState({
-        error: "Cannot move opponent's piece"
-      });
-    } else if (this.state.selected !== NOT_SELECTED) {
-      let captured = this.state.squares[index].player === this.opponentColor()
-      this.handleMove(captured, index);
-    }
-
-    // If the user selects a spot on the board that has a piece
-    else if (this.state.squares[index] != null) {
-      let newState = this.state.squares;
-      // Check whether or not that peice is already selected
-      if (this.state.squares[index].selected) {
-        newState[index].deselectPiece();
-        this.setState({
-          squares: newState,
-          selected: NOT_SELECTED,
-          error: ""
-        });
-      } else {
+    let newState = this.state.squares;
+    // if the player currently has no piece selected
+    if (this.state.selected === NOT_SELECTED) {
+      // if the spot that was clicked was the current player's color
+      if (this.state.squares[index].player === this.state.turn) {
+        this.checkPossibleMoves(index);
         newState[index].selectPiece();
         this.setState({
           squares: newState,
           selected: index,
           error: ""
         });
+      } else if (this.state.squares[index].player === this.opponentColor()) {
+        // if the spot that was clicked was the opponent's color
+        this.setState({
+          error: "Cannot move opponent's piece"
+        });
+      }
+    } else {
+      // if the player currently has a piece selected
+      // check the spot that was clicked
+
+      // if the spot that was clicked was the current player's color
+      if (this.state.squares[index].player === this.state.turn) {
+        // if the spot selected is the current selected piece, then we just need to
+        // deselect the piece
+        if (this.state.selected === index) {
+          this.dehighlightMoves();
+          newState[index].deselectPiece();
+          this.setState({
+            squares: newState,
+            selected: NOT_SELECTED,
+            error: ""
+          });
+        } else {
+          // if the spot selected is not the current selected piece throw an error,
+          // since you cannot capture your own piece
+          this.setState({
+            error: "Cannot cannot capture your own piece"
+          });
+        }
+      } else if (this.state.squares[index].player === this.opponentColor()) {
+        // if the spot that was clcked was the opposite player's color
+        this.handleMove(true, index);
+      } else if (this.state.squares[index].player === null) {
+        // if the spot that was clicked was an empty spot
+        this.handleMove(false, index);
```

Comment:
What does 'true' and 'false' here represent? Maybe we can wrap that in a variable like `isSelectionValid = true` and then call `this.handleMove(isSelectionValid, index)`